src/main/java/seedu/address/logic/commands/NegateCommand.java

```diff
@@ -0,0 +1,29 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.commons.core.index.Index;
+
+/**
+ * Represents a command with hidden internal logic and the ability to reverse the execution of other commands.
+ */
+public abstract class NegateCommand extends Command {
```

Comment:
does it have to be abstract since this class doesn't contain any abstract methods.. I think a normal class would suffice

Reply:
At first, I made it a non-abstract class, but then IntelliJ made noise because it extends abstract class Command so i either had to override the execute method in NegataCommand or make NegateCommand abstract. Since I personally felt that overriding the execute method in NegateCommand was a little redundant since the UnremindCommand, UnscheduleCommand and UndoneCommand will override the execute method again, I decided to make the NegateCommand class abstract.

I can change it back to a normal class if you want to though, I shall just override the abstract execute method in NegateCommand if you want to change it to a normal class :)

Reply:
I think just keep it as an abstract class will do.

Reply:
oh okay, I didn't know that Command was abstract. ya we can leave it as abstract then